emailsender/pkg/metrics/server_test.go

```diff
@@ -0,0 +1,49 @@
+package metrics
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type metricResponse map[string]*io_prometheus_client.MetricFamily
+
+func TestMetricsServerCorrectAddress(t *testing.T) {
+	server := NewMetricsServer(cfg)
+	defer func(server *http.Server) {
+		err := server.Close()
+		require.NoError(t, err)
+	}(server)
+	assert.Equal(t, ":9999", server.Addr)
+}
+
+func TestMetricsServerServesDefaultMetrics(t *testing.T) {
+	registry := initPrometheus(NewInstance())
+	metrics := serveMetrics(t, registry)
+	assert.Contains(t, metrics, "go_memstats_alloc_bytes", "not found default metrics")
+}
+
+func serveMetrics(t *testing.T, registry *prometheus.Registry) metricResponse {
+	rec := httptest.NewRecorder()
+	req, err := http.NewRequest(http.MethodGet, "/metrics", nil)
+	require.NoError(t, err, "failed creating metrics requests")
+
+	server := newMetricsServer(":9999", registry)
+	defer func(server *http.Server) {
+		err := server.Close()
+		require.NoError(t, err)
+	}(server)
+	server.Handler.ServeHTTP(rec, req)
+	require.Equal(t, http.StatusOK, rec.Code, "status code should be OK")
```

Comment:
Same here Listen was never called, so there's nothing to close we only use the internal Handler, which is kind of "simulating" the server without ever actually opening a network connection.
```suggestion
	server := newMetricsServer(":9999", registry)
	server.Handler.ServeHTTP(rec, req)
	require.Equal(t, http.StatusOK, rec.Code, "status code should be OK")
```